### PR TITLE
Adding docker compose env + pull requests template

### PR DESCRIPTION
**Pull request by eduardogr**
- Adding a way to run test locally without installing kubebuilder
  - using docker multistage build + docker-compose
  - so, now we can `docker-compose run tests` and have a clean running to test our code
  - production builds remain the same, but i added the possibility to manage it via docker-compose but keeping the same management of it within our Makefile

- Adding pull request template for future collaborations

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -9,11 +9,19 @@ jobs:
       KUBEBUILDER_CONTROLPLANE_START_TIMEOUT: "60s"
     steps:
     - checkout
+    - run:
+        name: Install Docker Compose
+        command: |
+          set -x
```
Comment (Wiston999) [inline .circleci/config.yml:15]: is the `set -x` really needed? It'll print all bash commands to be run.

File: Makefile [hunk 2]
```diff
@@ -13,7 +13,8 @@ all: manager
 
 # Run tests
 test: generate fmt vet manifests
-	go test -v ./backend... ./errors/... ./controllers/... -coverprofile cover.out
+	docker-compose run secrets-manager-local \
```
Comment (Wiston999) [inline Makefile:16]: I'd create a docker-compose task named `test` or similar that encapsulates this whole command. So you just `docker-compose run test` for testing everything

Comment (eduardogr) [reply Makefile]: i agree, rolling this back...

**Comment by eduardogr** [comment]
i totally forgot that binary, i've removed it and adding to gitignore

**Pull request merged**